src/index.ts

```diff
@@ -1,87 +1,80 @@
-import {API, APIClient, APIClientOptions, FetchProvider, UInt128} from '@wharfkit/antelope'
-import BN from 'bn.js'
+import {APIClient, Asset, Name, NameType} from '@wharfkit/antelope'
+import * as SystemTokenContract from './contracts/system.token'
 
-import {PowerUpAPI} from './powerup'
-import {RAMAPI} from './ram'
-import {REXAPI} from './rex'
-
-export * from './powerup'
-export * from './ram'
-export * from './rex'
-
-interface ResourcesOptions extends APIClientOptions {
-    api?: APIClient
-    sampleAccount?: string
-    symbol?: string
-    url?: string
-}
-
-export interface SampleUsage {
-    account: API.v1.AccountObject
-    cpu: UInt128
-    net: UInt128
+interface TokenOptions {
+    apiClient: APIClient
+    systemTokenSymbol?: Asset.SymbolType
 }
 
-export const BNPrecision = new BN(1000 * 1000)
+export class Token {
+    readonly systemTokenContract: SystemTokenContract.Contract
+    readonly systemTokenSymbol: Asset.SymbolType
 
-export class Resources {
-    static __className = 'Resources'
+    constructor({apiClient, systemTokenSymbol}: TokenOptions) {
+        this.systemTokenContract = new SystemTokenContract.Contract({client: apiClient})
+        this.systemTokenSymbol = systemTokenSymbol || '4,EOS'
+    }
 
-    readonly api: APIClient
+    async transfer(
+        from: NameType,
+        to: NameType,
+        amount: number,
+        memo = '',
+        symbolType: Asset.SymbolType = this.systemTokenSymbol
+    ) {
+        const symbol = Asset.Symbol.from(symbolType)
+        const senderBalance = await this.balance(from, symbol.code)
```

Comment:
Why are we checking the balance?

Reply:
The idea was to validate that the action can take place before carrying it out. I don't mind removing the balance checks though.

Reply:
I think we should probably remove it. Having it will result in two things:

- If there's an insufficient balance, it just replaces the `push_transaction` call with a `get_table_rows` call.
- If there's a balance, it does both, duplicating the number of API calls.

So it's never really reducing the API calls required to perform the action, and on success it's doubling them.